shell/app/electron_main_delegate.cc

```diff
@@ -281,6 +277,19 @@ void ElectronMainDelegate::PreSandboxStartup() {
   std::string process_type =
       command_line->GetSwitchValueASCII(::switches::kProcessType);
 
+  base::FilePath user_data_dir =
+      command_line->GetSwitchValuePath(::switches::kUserDataDir);
+  if (!user_data_dir.empty()) {
+    base::PathService::OverrideAndCreateIfNeeded(chrome::DIR_USER_DATA,
```

Comment:
How does this interact with things like https://github.com/electron/electron/blob/main/lib/browser/api/app.ts#L61-L72

Reply:
This is here because this needs to be correct in child processes. I think we might be able to delete (some of) that app.ts code now.

Reply:
Turns out we can delete all of it I'm pretty sure 👍